feat(gnovm/tests/stdlibs): add `fmt`

thehowl opened the pull request:
This PR removes `fmt` from being implemented using NativeValue / NativeType, and implements a mostly pure-Gno implementation which uses an initial, rudimentary implementation of Gno reflection for `fmt`.

This PR is ready for review, but some of its related work has been broken out in separate PR to aid reviewing and create atomic PRs.

- Blocks #3677
- Relates to #1361
- Depends on #3861
- Depends on #3862 
- Depends on #3875

## Summary of changes

### addition of `gnovm/tests/stdlibs/fmt`

The main addition is of this set of files, which comprises the bulk of the added lines of this PR. Most of this is a copy of Go's `fmt`. Changes are mostly in `errors.gno` and `format.gno` - the other files are left mostly intact.

At the end of `print.gno`, there is an added section containing a set of native functions which form the "reflection" needed to implement the `fmt` package in a manner that is similar to Go's. The implementation is not meant to be a public API and is meant to serve initially only for usage by `fmt`; we can use this as sketch to then implement a proper `reflect` package.

Attention in reviewing should mostly go to this `print.gno` and its `print.go` counterpart, especially for any missed cases or potential bugs.

### addition of `gnovm/tests/stdlibs/os`

Complementary to the above, to avoid some problems that were arising due to the handling of os.Stdin, Stdout and Stderr, I also removed the NativeValue / NativeType in `os` and replaced them with a native binding alternative.

To avoid having both an `Output` and `ErrorOutput` on the machine, this package supports an additional method on the `Machine.Output`, `StderrWrite`, which is implemented by writers who wish to differentiate between the two.

### other

- `gnovm/pkg/test`
	- added support for `StderrWrite` in its writers
	- removed native values for `os`  and `fmt`
	- combine `stdin`, `stdout`, `stderr` into a single `output` parameter in the related functions.
- `gnovm/pkg/gnolang`
	- support testing `gnovm/tests/stdlibs`, by only testing stdlibs that exist there and not in the official stdlibs.
- `gnovm/cmd/gno`
	- fix tests which needed the `fmt` package.
- `gnovm/tests/files`
	- most changed tests are result of the improved functionality of the `fmt` package, allowing for instance to show declared types correctly.
- `gnovm/stdlibs/bufio`
	- due to the removal of `os.Stdin`, this example had to be changed.
- `misc/genstd`
	- added a flag `-skip-init-order` when creating it could cause problems, like the testing stdlibs; fixed a bug when using `TypedValue`.

Gno2D2 commented:
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 The pull request was created from a fork (head branch repo: thehowl/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

thehowl commented:
This mostly works. TODO:

- split PR: adding `any` type to uverse
- split PR: fix dereference calling a non-pointer method on a nil-pointer value
- Fixup `fmt` tests to work, ie. no reflection and no dot-imports.

ltzmaxwell reviewed (approved):
After comparing with the native Go fmt, it follows the original design, and the mini reflection implemented through nativebinding is cleverly designed. LGTM.

ltzmaxwell commented on gnovm/tests/files/types/eql_0f35.gno:
```diff
@@ -31,4 +31,4 @@ func main() {
 // Output:
 // Recovered. Error:
-//  0
+//  error: 0
```
is there an extra space here?

ltzmaxwell replied:
that's what fmt.Println does. ok.

mvertes reviewed (approved):
LGTM

thehowl commented:
> Incidentally, I think that this PR may also solve #3687, through implementation of `X_getAddr()` in `gnovm/tests/sdtlibs/fmt/print.go` although I haven't yet verified it.

That issue is still valid for `println()`, and the implementation of `getAddr` is actually non-deterministic; but it's fine because `fmt` can only be used off-chain.

When we move `fmt` on-chain (or better, when we make a proper `reflect` implementation), I'd like to figure out a way to make a proper deterministic `getAddr` implementation.
